crates/hyperqueue/src/dashboard/ui/screen.rs

```diff
@@ -1,10 +1,47 @@
+use tako::WorkerId;
 use termion::event::Key;
 
 use crate::dashboard::data::DashboardData;
 use crate::dashboard::ui::terminal::DashboardFrame;
 
 pub trait Screen {
     fn draw(&mut self, frame: &mut DashboardFrame);
-    fn update(&mut self, data: &DashboardData);
-    fn handle_key(&mut self, key: Key);
+    fn update(&mut self, data: &DashboardData, controller: &mut ScreenController);
+    fn handle_key(&mut self, key: Key, controller: &mut ScreenController);
+}
+
+#[derive(Clone, Copy)]
+pub enum DashboardScreenState {
+    ClusterOverviewScreen,
+    WorkerOverviewScreen(Option<WorkerId>),
```

Comment:
The individual screens already hold their inner state, so I don't think that it's correct to duplicate that state here. It would be useful if we recreated the screens from scratch everytime, but we decided that we don't want to do that. So this should be just a simple enum.